homeassistant/components/fritz/services.py

```diff
@@ -0,0 +1,62 @@
+"""Services for Fritz integration."""
+import logging
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.service import async_extract_config_entry_ids
+
+from .const import DOMAIN, FRITZ_SERVICES, SERVICE_REBOOT, SERVICE_RECONNECT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_services(hass: HomeAssistant):
+    """Set up services for Fritz integration."""
+    if hass.data.get(FRITZ_SERVICES, False):
+        return
+
+    hass.data[FRITZ_SERVICES] = True
+
+    async def async_call_fritz_service(service_call):
+        """Call correct Fritz service."""
+
+        if not (
+            fritzbox_entry_ids := await _async_get_configured_fritz_tools(
+                hass, service_call
+            )
+        ):
+            raise HomeAssistantError(
+                f"Failed to call service '{service_call.service}'. Config entry for target not found"
+            )
+
+        for entry in fritzbox_entry_ids:
```

Comment:
Should we use asyncio.gather here?

Reply:
Actually no since it's generating executor jobs.

Reply:
We would save some context switches if we would wrap the iteration inside a function and make the service helper sync.